ex5/dist/main.js

```diff
@@ -0,0 +1,156 @@
+class Main {
+	constructor() {
+		this.itemClient = new ItemClient();
+	}
+
+	init = async () => {
+		const addItemButton = document.getElementById('list-item-submit');
+		addItemButton.addEventListener('click', this.handleItem);
+
+		const deleteAllBtn = document.getElementById('list-item-clear');
+		deleteAllBtn.addEventListener('click', this.handleDeleteAll);
+
+		await this.renderItems();
+	};
+
+	handleItem = async () => {
+		const input = document.getElementById('list-item-input');
+		const inputValue = input.value;
+		await this.itemClient.postItem(inputValue);
+		await this.renderItems();
+	};
+
+	deleteItem = async (item) => {
+		await this.itemClient.deleteItem(item);
+		await this.renderItems();
+	};
+	popUp = async (item) => {
+		const { value: text } = await swal.fire({
+			input: 'textarea',
+			inputLabel: 'Edit your task',
+			inputValue: item,
+			inputAttributes: {
+				'aria-label': 'Type your message here',
+			},
+			showCancelButton: true,
+		});
+		if (text) {
+			await this.itemClient.editItem(item, text);
+		}
+	};
+	editItem = async (item) => {
+		let itemToEdit;
+		const list = document.querySelectorAll('.list-item');
+		for (const listItem of list) {
+			if (listItem.textContent == item) {
+				itemToEdit = listItem;
+				await this.popUp(item);
+			}
+		}
+
+		await this.renderItems();
+	};
+
+	updateStatus = async (item) => {
+		await this.itemClient.updateStatus(item);
+	};
+
+	renderItems = async () => {
+		const list = document.getElementById('list');
+		list.innerHTML = '';
+		const items = await this.itemClient.getItems();
+
+		items.forEach((item) => {
+			const listItem = document.createElement('li');
+			listItem.classList.add('list-item');
+			const listItemStatusButton = this._createStatusButton(
+				item.ItemName,
+				item.status
+			);
+			const listItemDeleteButton = this._createDeleteButton(
+				item.ItemName
+			);
+			const editItemButton = this._createEditButton(item.ItemName);
+
+			listItem.appendChild(listItemStatusButton);
+			const span = document.createElement('span');
+			span.innerHTML = item.ItemName;
+			span.className = 'task-content';
+
+			listItem.appendChild(span);
+			listItem.appendChild(editItemButton);
+			listItem.appendChild(listItemDeleteButton);
+			list.appendChild(listItem);
+		});
+		this.updateCounter(items);
+		this.changeImageState();
+		this.updateTaskCounter();
+		this.clearForm();
+	};
+
+	handleDeleteAll = async () => {
+		const items = await this.itemClient.getItems();
+		items.forEach((item) => {
+			this.deleteItem(item.ItemName);
```

Comment:
bug:
since deleteItem is async, this method exits before the deletion is done, and then tries to render the list again
sometimes if you click delete all - it will not delete everything from the UI.
This also happens when catching multiple pokemon - try to enter 1,2,3,4,5 in the input - it will still display an empty list